src/Simplex/Chat/Store.hs

```diff
@@ -380,81 +394,99 @@ getGroup st User {userId, userContactId} localDisplayName =
-            JOIN groups g ON g.group_id = m.group_id
             JOIN contact_profiles p ON p.contact_profile_id = m.contact_profile_id
-            JOIN connections c ON c.group_member_id = m.group_member_id
-            WHERE g.group_id = ?
-            ORDER BY c.connection_id DESC
-            LIMIT 1
+            LEFT JOIN connections c ON c.connection_id = (
+              SELECT max(cc.connection_id)
+              FROM connections cc
+              where cc.group_member_id = c.group_member_id
+            )
+            WHERE m.group_id = ?
           |]
           (Only groupId)
-    getUserMember_ :: DB.Connection -> Int64 -> ExceptT StoreError IO GroupMember
-    getUserMember_ db groupId = ExceptT $ do
-      userMember
-        <$> DB.query
-          db
-          [sql|
-            SELECT
-              m.group_member_id, m.member_id, m.member_role, m.member_status,
-              m.invited_by, m.contact_id, p.display_name, p.full_name
-            FROM group_members m
-            JOIN groups g ON g.group_id = m.group_id
-            JOIN contact_profiles p ON p.contact_profile_id = m.contact_profile_id
-            WHERE g.group_id = ? AND m.contact_id = ?
-          |]
-          (groupId, userContactId)
-    toContactMember :: (GroupMemberRow :. ConnectionRow) -> (GroupMember, Connection)
-    toContactMember (memberRow :. connRow) = (toGroupMember memberRow, toConnection connRow)
-    toGroupMember :: GroupMemberRow -> GroupMember
-    toGroupMember (groupMemberId, memberId, memberRole, memberStatus, invitedById, memberContactId, displayName, fullName) =
-      let memberProfile = Profile {displayName, fullName}
-          invitedBy = toInvitedBy userContactId invitedById
-       in GroupMember {groupMemberId, memberId, memberRole, memberStatus, invitedBy, memberProfile, memberContactId}
-    userMember :: [GroupMemberRow] -> Either StoreError GroupMember
-    userMember [memberRow] = Right $ toGroupMember memberRow
-    userMember _ = Left SEGroupWithoutUser
+    toContactMember :: (GroupMemberRow :. MaybeConnectionRow) -> (GroupMember, Maybe Connection)
+    toContactMember (memberRow :. connRow) = (toGroupMember userContactId memberRow, toMaybeConnection connRow)
+    splitUserMember_ :: [(GroupMember, Maybe Connection)] -> Either StoreError ([(GroupMember, Maybe Connection)], GroupMember)
+    splitUserMember_ allMembers =
+      let (b, a) = break ((== Just userContactId) . memberContactId . fst) allMembers
+       in case a of
+            [] -> Left SEGroupWithoutUser
+            u : ms -> Right (b <> ms, fst u)
+
+getGroupInvitation :: StoreMonad m => SQLiteStore -> User -> GroupName -> m ReceivedGroupInvitation
+getGroupInvitation st user localDisplayName =
+  liftIOEither . withTransaction st $ \db -> runExceptT $ do
+    (Group {membership, members, groupProfile}, qInfo) <- getGroup_ db user localDisplayName
+    when (memberStatus membership /= GSMemInvited) $ throwError SEGroupAlreadyJoined
```

Comment:
this probably should be checked on chat level, not store - it's only a get method

Reply:
I did it on purpose to not depend on the decision that group invitations are stored as groups - internally a different record is used. Hence the check here.